### PR TITLE
Test/integration test

### PR DESCRIPTION
**Pull request by esherman-credo**
## Describe your changes
Implementation of a test for end-to-end functionality of Platform -> Connect -> Lens -> Connect -> Platform. It relies on the DEV version of the platform (rather than QA/Prod).

This test currently supports 11 evidence requirements from 6 different evaluators: Performance, DataProfiler, ModelFairness, Security, FeatureDrift, and ModelEquity.

This is currently a DRAFT PR --> the test relies on an API token and an API url from the Platform. These will be specified in the functions added to the bottom of `conftest.py` but they cannot be stored in cleartext. @credo-nate and I will work together to set up the token and url as Github secrets.

## Issue ticket number and link
N/A

## Known outstanding issues that are not fully accounted for
There are several evaluators in Lens that could not be immediately added to this test for various reasons. These include:
--DataFairness/DataEquity --> Platform does not presently support evidences which do not have a `model_name` tag in the metadata. These evaluators are inherently model-independent and supporting them will require a (likely minor) change to requirements matching.
--Privacy/DataFairness --> See https://github.com/credo-ai/credoai_lens/issues/265

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have built basic tests for new functionality (particularly new evaluators)
-- N/A this is a test
- [x] If new libraries have been added, I have checked that [readthedocs](https://readthedocs.org/projects/credoai-lens/) API documentation is constructed correctly
-- N/A
- [x] Will this be part of a major product update? If yes, please write one phrase about this update.
-- No

## Extra-mile Checklist
- [x] I have thought expansively about edge cases and written tests for them

### CODE REVIEW

**Review by IanAtCredo — APPROVED** [review]
This looks to still be WIP, but looks good to me.

File: credoai/lens/pipeline_creator.py [hunk 1]
```diff
@@ -58,7 +59,7 @@ def extract_metrics(labels):
     """Extract metrics from a single evidence requirement"""
     metrics = set()
     if "metric_type" in labels:
-        metrics.add(labels["metric_type"].removesuffix("_parity"))
```
[the remark this answers is not in the document]
Comment (esherman-credo) [reply credoai/lens/pipeline_creator.py:61]: See conversation starting [here](https://github.com/credo-ai/credoai_lens/pull/279#issuecomment-1338696284)

To summarize, Nate was getting an error. The `removesuffix` standard library function was only introduced in Py3.9 (I usually test with 3.10+ so I've never run into this issue), while Lens only requires Py3.8.

He switched it to the version in [utils.common](https://github.com/credo-ai/credoai_lens/blob/develop/credoai/utils/common.py#L107)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -5,6 +5,7 @@ on:
     branches:
       - main
       - develop
+      - test/integration_test
```
[the remark this answers is not in the document]
Comment (esherman-credo) [reply .github/workflows/test.yml:8]: Yeah. Let's keep it in until just before merge to make sure test continues to run.

[the remark this answers is not in the document]
Comment (esherman-credo) [reply .github/workflows/test.yml:8]: Ah I see you approved already, nevermind. Cutting that line and merging.
